### DISCUSSION TITLE
Proxies between MCP Clients and MCP Servers

### DISCUSSION THREAD

**Discussion by LaurentAjdnik** [modelcontextprotocol/modelcontextprotocol#1134] [Ideas - General]
### Pre-submission Checklist

- [X] I have verified this would not be more appropriate as a feature request in a specific repository
- [X] I have searched existing discussions to avoid duplicates

### Your Idea

Hello everyone!

I've had this crazy idea of proxies between MCP Clients and MCP Servers.

This would allow a bunch of new features, while being totally MCP-compliant.

For instance:
- Logging messages
- Blocking capabilities
- Multiplexing MCP Servers
- Translating content on-the-fly
- And many others...
 
I drafted a few principles [here](https://github.com/LaurentAjdnik/mcproxy).

Does it make any sense to you? At least, it does a lot to me. :grin:

Feedback would be greatly appreciated!

To keep discussions centralized, you can react [here](https://github.com/LaurentAjdnik/mcproxy/discussions/2). Thanks!

### Scope

- [ ] Protocol Specification
- [ ] SDK Features
- [ ] Documentation
- [X] Developer Experience
- [X] Other

**Comment by shenli** [comment]
https://github.com/lightconetech/mcp-gateway I implement a mcp proxy to allow Claude desktop talk to remote MCP servers. This is the same architecture with the "simplest form" in your proposal.

**Reply by LaurentAjdnik** [reply]
Nice use case! Thanks for sharing!

**Comment by ggoodman** [comment]
I wrote some additional thoughts on this in the specification repo's discussions: https://github.com/modelcontextprotocol/specification/discussions/94. PTAL!

**Comment by localden** [comment]
We've limited Discussions in this repository to meeting notes and are closing this thread. To continue the conversation, pick a channel from [Contributor Communication](https://modelcontextprotocol.io/community/communication). Thanks for posting.
